packages/api/src/controllers/access-control.test.ts

```diff
@@ -0,0 +1,175 @@
+
+// includes auth file tests
+
+let server: TestServer;
+let mockAdminUserInput: User;
+let mockNonAdminUserInput: User;
+
+// jest.setTimeout(70000)
+
+beforeAll(async () => {
+  server = await serverPromise;
+
+  mockAdminUserInput = {
+    email: "user_admin@gmail.com",
+    password: "x".repeat(64),
+  };
+
+  mockNonAdminUserInput = {
+    email: "user_non_admin@gmail.com",
+    password: "y".repeat(64),
+  };
+});
+
+afterEach(async () => {
+  await clearDatabase(server);
+});
+
+describe("controllers/signing-key", () => {
+  describe("basic CRUD with JWT authorization", () => {
+    let client: TestClient;
+    let adminUser: User;
+    let adminToken: string;
+    let nonAdminUser: User;
+    let nonAdminToken: string;
+    let otherUserToken: string;
+    let signingKey: WithID<SigningKey>;
+    let gatedPlaybackId: string;
+    let publicPlaybackId: string;
+    let gatedAsset: WithID<Asset>;
+
+    beforeEach(async () => {
+      ({ client, adminUser, adminToken, nonAdminUser, nonAdminToken } =
+        await setupUsers(server, mockAdminUserInput, mockNonAdminUserInput));
+      client.jwtAuth = nonAdminToken;
+      let created: SigningKeyResponsePayload = await client
+        .post("/access-control/signing-key")
+        .then((res) => res.json());
+      let res = await client.get(`/access-control/signing-key/${created.id}`);
+      signingKey = await res.json();
+      const res2 = await client.post("/stream", {
+        name: "test",
+        playbackPolicy: {
+          type: "signed",
+        },
+      });
+      const stream = await res2.json();
+      gatedPlaybackId = stream.playbackId;
+      const res3 = await client.post("/stream", {
+        name: "test",
+        playbackPolicy: {
+          type: "public",
+        },
+      });
+      const publicStream = await res3.json();
+      publicPlaybackId = publicStream.playbackId;
+      let id = uuid();
+      gatedAsset = await db.asset.create({
+        id,
+        name: "test-storage",
+        createdAt: Date.now(),
+        playbackId: await generateUniquePlaybackId(id),
+        playbackPolicy: {
+          type: "signed",
+        },
+        status: {
+          phase: "ready",
+          updatedAt: Date.now(),
+        },
+        userId: nonAdminUser.id,
+      });
+    });
+
+    it("should create a gate stream and allow playback with given public key", async () => {
+      client.jwtAuth = adminToken;
+      const res2 = await client.post("/access-control/gate", {
+        stream: `video+${gatedPlaybackId}`,
+        pub: signingKey.publicKey,
+        type: "jwt",
+      });
+
+      expect(res2.status).toBe(204);
+    });
+
+    it("should not allow playback on not existing streams or public keys", async () => {
+      const res = await client.post("/access-control/gate", {
+        stream: `video+0000000000`,
+        pub: signingKey.publicKey,
+        type: "jwt",
+      });
+      expect(res.status).toBe(403);
```

Comment:
Should these be 404s instead?

Reply:
Could be, in the spec doc and the discussion with Redouane we defined that the possible Status Codes are `200` for allowed playback (became 204) and `403` for disallowed playback, but we could just do 2XX and 4XX so we can be more specific!